File: AccelerationToMulitcore/1.CPUOpenmp/00.ParallelExecution/step2/HelloWorld.c
#include <stdio.h>
#include <omp.h> 

int main(int argc, char* argv[])
{

 int tid, nthreads; 

#pragma omp parallel private(tid)
{
  /* Get thread ID */
  tid=omp_get_thread_num();
  /* Get number of threads */
  nthreads = omp_get_num_threads();

  printf("Hello, I am thread id %d of %d.\n ", tid, nthreads);

}
  return 0;
}

File: AccelerationToMulitcore/1.CPUOpenmp/00.ParallelExecution/step2/HelloWorld.f90
! =========================
!  Get thread no. and id
! =========================

program helloWorld

      use omp_lib

      implicit none
      integer :: tid, nthreads

      !$omp parallel private(tid)

      !Get and print thread id
      tid = OMP_GET_THREAD_NUM()
      nthreads = OMP_GET_NUM_THREADS()
      
      print *, "Hello World! I'm tid", tid, "in nthreads", nthreads
          
      !$omp end parallel 
end program helloWorld
